packages/utils/ast-utils.ts

```diff
@@ -0,0 +1,567 @@
+		);
+	}
+}
+
+/**
+ *
+ * Find paths that match `new name.space.PluginName()` for a
+ * given array of plugin names
+ *
+ * @param {any} j — jscodeshift API
+ * @param {Node} node - Node to start search from
+ * @param {String[]} pluginNamesArray - Array of plugin names like `webpack.LoaderOptionsPlugin`
+ * @returns {Node} Node that has the pluginName
+ */
+
+function findPluginsByName(j: any, node: INode, pluginNamesArray: string[]): INode {
+	return node
+		.find(j.NewExpression)
+		.filter((path: INode): boolean => {
+			return pluginNamesArray.some(
+				(plugin: string) =>
+					memberExpressionToPathString(path.get("callee").value) === plugin,
+			);
+	});
+}
+
+/**
+ * It lookouts for the plugins property and, if the array is empty, it removes it from the AST
+ * @param  {any} j - jscodeshift API
+ * @param  {Node} rootNode - node to start search from
+ * @returns {Node} rootNode modified AST.
+ */
+
+function findPluginsArrayAndRemoveIfEmpty(j: any, rootNode: INode): INode {
+	return rootNode
+		.find(j.Identifier, { name: "plugins" })
+		.forEach((node: INode) => {
+			const elements: INode[] = safeTraverse(node, [
+				"parent",
+				"value",
+				"value",
+				"elements",
+			]);
+			if (!elements.length) {
+				j(node.parent).remove();
+			}
+		});
+}
+
+/**
+ *
+ * Finds the path to the `name: []` node
+ *
+ * @param {any} j — jscodeshift API
+ * @param {Node} node - Node to start search from
+ * @param {String} propName - property to search for
+ * @returns {Node} found node and
+ */
+
+function findRootNodesByName(j: any, node: INode, propName: string): INode {
+	return node.find(j.Property, { key: { name: propName } });
+}
+
+/**
+ *
+ * Creates an Object's property with a given key and value
+ *
+ * @param {any} j — jscodeshift API
+ * @param {String | Number} key - Property key
+ * @param {String | Number | Boolean} value - Property value
+ * @returns {Node}
+ */
+
+function createProperty(j: any, key: string | number, value: string | number | boolean): any {
+	return j.property(
+		"init",
+		createIdentifierOrLiteral(j, key),
+		createLiteral(j, value),
+	);
+}
+
+/**
+ *
+ * Creates an appropriate literal property
+ *
+ * @param {any} j — jscodeshift API
+ * @param {String | Boolean | Number} val
+ * @returns {Node}
+ */
+
+function createLiteral(j: any, val: string | boolean | number): any {
+	let literalVal: string | boolean | number = val;
+	// We'll need String to native type conversions
+	if (typeof val === "string") {
+		// 'true' => true
+		if (val === "true") { literalVal = true; }
+		// 'false' => false
+		if (val === "false") { literalVal = false; }
+		// '1' => 1
+		if (!isNaN(Number(val))) { literalVal = Number(val); }
+	}
+	return j.literal(literalVal);
+}
+
+/**
+ *
+ * Creates an appropriate identifier or literal property
+ *
+ * @param {any} j — jscodeshift API
+ * @param {String | Boolean | Number} val
+ * @returns {Node}
+ */
+
+function createIdentifierOrLiteral(j: any, val: any): any {
```

Comment:
This is the same as you provided earlier, `string`, `array` etc..

Reply:
Where exactly are you pointing? Arguments or return value?

Reply:
both, the val could you create a interface type for and return value is a node

Reply:
not in all cases, but one of many, please be more specific here